pypicloud/cache/dynamo.py

```diff
@@ -22,6 +25,78 @@
 LOG = logging.getLogger(__name__)
 
 
+def _path_depth(package):
+    """Count the slashes in path, or return infinity on no path."""
+    path = package.data.get('path')
+    if not path:
+        return float('+inf')
+    return path.count('/')
+
+
+def _decide_between_versions(contender, current):
+    """Decide between packages with the same filename on different paths."""
+
+    if contender.last_modified < current.last_modified:
+        return contender
```

Comment:
Ah, I was confused because I thought the goal was to get the freshest data. I'm totally behind you on making the behavior be the "least surprising". That seems very reasonable.

Thinking about it some more, if you're trying to make it the least surprising, wouldn't you just ignore files with the same filename that are in both cache & storage? I guess it would only cause the situation you mentioned if the paths were different. So I guess the gain from `_decide_between_versions` is that if you have the same filename in storage & cache, *but* the one in storage has a different `last_modified` (and the same `path`), then the `last_modified` in the cache will be updated.

I think it may be best to just take the one already in the cache if the filenames are equal. If you do want to keep this then I think you have to change it to first return the cached package if the paths are not equal, and then do any other comparisons after that.